Sources_v3/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -164,11 +172,22 @@ open class ChatChannelVC<ExtraData: UIExtraDataTypes>: ViewController,
     }
     
     // MARK: - UIScrollViewDelegate
+
+    public func scrollViewWillEndDragging(
+        _ scrollView: UIScrollView,
+        withVelocity velocity: CGPoint,
+        targetContentOffset: UnsafeMutablePointer<CGPoint>
+    ) {
+        if targetContentOffset.pointee.y < collectionViewLayout.zeroOffset.y {
+            targetContentOffset.pointee.y = collectionViewLayout.zeroOffset.y
+        }
+    }
```

Comment:
Is there any chance how to do this from within the layout itself?

Reply:
Actually there is! 
Moved almost all logic inside.
Only thing we cannot prevent is to check offset for message preloading. But preloading can be done differently in general (via `willShowItemAtIndexPath`) so I let it be.

Reply:
Very nice! I think we can live with having it done like this for now. Thank you!